libeos-updater-util/avahi-service-file.c

```diff
@@ -110,35 +179,35 @@ generate_avahi_service_template_to_file (GFile *path,
                                   error);
 }
 
-static gchar *
-txt_record (const gchar *key,
-            const gchar *value)
-{
-  return g_strdup_printf ("%s=%s", key, value);
-}
-
 static gboolean
 generate_v1_service_file (const gchar *ostree_path,
                           GDateTime *head_commit_timestamp,
                           GFile *service_file,
                           GCancellable *cancellable,
                           GError **error)
 {
-  g_autoptr(GPtrArray) txt_records = NULL;
   g_autofree gchar *timestamp_str = NULL;
+  const GVariantType *records_type = G_VARIANT_TYPE ("a(sv)");
+  g_auto(GVariantBuilder) records_builder = G_VARIANT_BUILDER_INIT (records_type);
+  g_autoptr(GVariant) txt_records = NULL;
 
   timestamp_str = g_date_time_format (head_commit_timestamp, "%s");
-  txt_records = g_ptr_array_new_with_free_func (g_free);
-
-  g_ptr_array_add (txt_records, txt_record (eos_avahi_v1_ostree_path,
-                                            ostree_path));
-  g_ptr_array_add (txt_records, txt_record (eos_avahi_v1_head_commit_timestamp,
-                                            timestamp_str));
+  g_variant_builder_add (&records_builder, "(sv)",
+                         "eos_txt_version",
+                         g_variant_new_string ("1"));
+  g_variant_builder_add (&records_builder, "(sv)",
+                         eos_avahi_v1_ostree_path,
+                         g_variant_new_string (ostree_path));
+  g_variant_builder_add (&records_builder, "(sv)",
+                         eos_avahi_v1_head_commit_timestamp,
+                         g_variant_new_string (timestamp_str));
+  txt_records = g_variant_ref_sink (g_variant_builder_end (&records_builder));
```

Comment:
I'd say no, even if it means that we need to worry about sinking the gvariant ourselves and general verbosity of the code. My impression is that functions that actually store the gvariant should call `ref_sink`. In our case the functions simply read contents of the gvariant.